src/mesh/boundary_info.C

```diff
@@ -1505,38 +1423,31 @@ void BoundaryInfo::build_node_list (std::vector<dof_id_type>& nl,
 void
 BoundaryInfo::build_node_list_from_side_list()
 {
-  std::multimap<const Elem*,
-                std::pair<unsigned short int,
-                          boundary_id_type> >::const_iterator pos;
-
-  //Loop over the side list
-  for (pos=_boundary_side_id.begin(); pos != _boundary_side_id.end(); ++pos)
+  // Loop over the side list
+  boundary_side_iter pos = _boundary_side_id.begin();
+  for (; pos != _boundary_side_id.end(); ++pos)
     {
       // Don't add remote sides
-      if(pos->first->is_remote())
+      if (pos->first->is_remote())
         continue;
 
-      //Need to loop over the sides of any possible children
-      std::vector< const Elem * > family;
+      // Need to loop over the sides of any possible children
+      std::vector<const Elem *> family;
 #ifdef LIBMESH_ENABLE_AMR
       pos->first->active_family_tree_by_side (family, pos->second.first);
 #else
       family.push_back(pos->first);
 #endif
 
-      for(std::size_t elem_it=0; elem_it < family.size(); elem_it++)
+      for (std::size_t elem_it=0; elem_it < family.size(); elem_it++)
         {
           const Elem * cur_elem = family[elem_it];
 
           UniquePtr<Elem> side = cur_elem->build_side(pos->second.first);
 
-          //Add each node node on the side with the side's boundary id
-          for(unsigned int i=0; i<side->n_nodes(); i++)
-            {
-              Node * node = side->get_node(i);
-
-              this->add_node(node, pos->second.second);
-            }
+          // Add each node node on the side with the side's boundary id
+          for (unsigned int i=0; i<side->n_nodes(); i++)
+            this->add_node(side->get_node(i), pos->second.second);
```

Comment:
Don't *always* try to squeeze code like this into one-liners.  There's probably at least a dozen places where we use a temporary variable because I knew that opt mode would optimize it out but I found it useful to have a name to query in dbg/devel mode from gdb.

Obviously this isn't one of those cases; functions with no side effects like get_node() are easy to call from gdb.  But the other cases superficially look the same.  ;-)